rh-summit-2018/module-03.adoc

```diff
@@ -1,3 +1,302 @@
-== Running hiking demo with Kafka Connect and Debezium
+== Change Data Capture with Kafka Connect and Debezium
 
-TBD
\ No newline at end of file
+This part of the lab introduces you to change data capture (CDC) using http://debezium.io/[Debezium].
+Debezium allows you to capture data changes from MySQL, Postgres and MongoDB and stream those events into Apache Kafka.
+It is based on Kafka Connect.
+
+In the following you'll learn the following things:
+
+* Setting up Kafka Connect with the Debezium CDC connectors
+* Setting up the Debezium MySQL connector, ingesting changes from an existing example CRUD application
+* Streaming Data Changes to a Postgres Sink Database using the Confluent JDBC sink connector
+* Setting up a WildFly Swarm application for consuming the same CDC events and stream them to a web browser via WebSockets
+
+=== Setting Up the Example Application
+
+We've prepared a small CRUD application which we'll use in the following as source of data change events.
+It is a Java EE application running on WildFly and using MySQL as its database.
+
+Start a MySQL instance by running:
+
+[source]
+oc new-app --name=mysql debezium/example-mysql:0.7
+oc env dc/mysql MYSQL_ROOT_PASSWORD=debezium MYSQL_USER=mysqluser MYSQL_PASSWORD=mysqlpw
+
+Then build and start the application.
+This uses the https://github.com/openshift-s2i/s2i-wildfly[WildFly S2I] ("source to image") builder image which will fetch the application's source code from the given location,
+build it using Maven and deploy it to WildFly, setting up a MySQL datasource using the given credentials:
+
+[source]
+oc new-app --name=hikingapp openshift/wildfly-120-centos7:latest~https://github.com/strimzi/strimzi-lab.git \
+    --context-dir=hiking-demo \
+    -e MYSQL_DATABASE=inventory \
+    -e MYSQL_PASSWORD=mysqlpw \
+    -e MYSQL_USER=mysqluser \
+    -e MYSQL_DATASOURCE=HikingDS
+oc expose svc hikingapp
+
+Verify that the pods of the database and the application are running:
+
+[source]
+oc get pods
+
+Access the application in your web browser:
+
+[source]
+http://hikingapp-default.<YOUR_IP>.xip.io/hikr-1.0-SNAPSHOT/hikes.html
+
+You can use "Populate Data" to create some example data.
+Otherwise just insert a few records as you like.
+
+=== Setting Up Kafka Connect With the Debezium Connectors
+
+Deploy Kafka via Strimzi (TODO: should already be done as part of module 1):
+
+[source]
+----
+export STRIMZI_VERSION=0.2.0
+git clone -b $STRIMZI_VERSION https://github.com/strimzi/strimzi
+cd strimzi
+
+oc create -f examples/install/cluster-controller && oc create -f examples/templates/cluster-controller
+
+oc new-app strimzi-ephemeral -p CLUSTER_NAME=broker -p ZOOKEEPER_NODE_COUNT=1 -p KAFKA_NODE_COUNT=1 -p KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR=1 -p KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR=1
+----
+
+Deploy Kafka Connect With Debezium:
+
```

Comment:
Should we explain better what we are going to do so what is the `strimzi-connect-s2i` template and how it allows the users to add new plugins using the following build ?

Reply:
Yes, I've planned to expand instructions here and in other places.